docs/src/main/tut/typeclasses/traverse.md

```diff
@@ -5,235 +5,138 @@ section: "typeclasses"
-trivially satisfy the `F[_]` shape required by `Applicative`.
+As it turns out every `Traverse` is a lawful `Functor`. By carefully picking the `G` to
+use in `traverse` we can implement `map`.
 
-```tut:silent
-import cats.Semigroup
-import cats.data.{NonEmptyList, OneAnd, Validated, ValidatedNel}
-import cats.implicits._
+First let's look at the two signatures.
 
-def parseIntEither(s: String): Either[NumberFormatException, Int] =
-  Either.catchOnly[NumberFormatException](s.toInt)
+```tut:book:silent
+def traverse[F[_]: Traverse, G[_]: Applicative, A, B](fa: F[A])(f: A => G[B]): G[F[B]] = ???
 
-def parseIntValidated(s: String): ValidatedNel[NumberFormatException, Int] =
-  Validated.catchOnly[NumberFormatException](s.toInt).toValidatedNel
+def map[F[_]: Traverse, A, B](fa: F[A])(f: A => B): F[B] = ???
 ```
 
-Examples.
-
-```tut:book
-val x1 = List("1", "2", "3").traverseU(parseIntEither)
-val x2 = List("1", "abc", "3").traverseU(parseIntEither)
-val x3 = List("1", "abc", "def").traverseU(parseIntEither)
+Both have an `F[A]` parameter and a similar `f` parameter. `traverse` expects the return type
+of `f` to be `G[B]` whereas `map` just wants `B`. Similarly the return type of `traverse` is
+`G[F[B]]` whereas for `map` it's just `F[B]`. This suggests we need to pick a `G` such that
+`G[A]` communicates exactly as much information as `A`. We can conjure one up by simply wrapping
+an `A`.
 
-val v1 = List("1", "2", "3").traverseU(parseIntValidated)
-val v2 = List("1", "abc", "3").traverseU(parseIntValidated)
-val v3 = List("1", "abc", "def").traverseU(parseIntValidated)
+```tut:book:silent
+final case class Id[A](value: A)
 ```
 
-Notice that in the `Either` case, should any string fail to parse the entire traversal
-is considered a failure. Moreover, once it hits its first bad parse, it will not
-attempt to parse any others down the line (similar behavior would be found with
-using `Option` as the effect). Contrast this with `Validated` where even
-if one bad parse is hit, it will continue trying to parse the others, accumulating
-any and all errors as it goes. The behavior of traversal is closely tied with the
-`Applicative` behavior of the data type.
+In order to call `traverse` `Id` needs to be `Applicative` which is straightforward - note that while
+`Id` just wraps an `A`, it is still a type constructor which matches the shape required by `Applicative`.
 
-Going back to our `Future` example, we can write an `Applicative` instance for
-`Future` that runs each `Future` concurrently. Then when we traverse a `List[A]`
-with an `A => Future[B]`, we can imagine the traversal as a scatter-gather.
-Each `A` creates a concurrent computation that will produce a `B` (the scatter),
-and as the `Future`s complete they will be gathered back into a `List`.
+```tut:book:silent
+implicit val applicativeForId: Applicative[Id] = new Applicative[Id] {
+  def ap[A, B](ff: Id[A => B])(fa: Id[A]): Id[B] = Id(ff.value(fa.value))
 
-### Playing with `Reader`
```

Comment:
I found this section is another good example of `Reader`, how about we move it the [kleisli.md](https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/Traverse.scala) instead of losing it?